lib/snail.rb

```diff
@@ -89,12 +89,16 @@ def origin
     @origin ||= Snail.home_country
   end
 
-  def to_s
-    [name, line_1, line_2, city_line, country_line].select{|line| !(line.nil? or line.empty?)}.join("\n")
+  def to_s(country: false)
+    address_lines = [name, line_1, line_2, city_line]
+    if country || (@country && self.origin != @country)
```

Comment:
how about extracting the `origin != @country` logic to something like `def international?`?